### DISCUSSION TITLE
Attachments not showing up in data tab

### DISCUSSION THREAD

**Discussion by nunopoco** [Budibase/budibase#7777] [Help] [answered]
Hi everyone,

I need help as my attachments don't show up when adding them directly in the data tab, by creating a new row.

I've installed Budibase in a DigitalOcean droplet using the 1-click process and noticed that attachments are not working. I saw [on the docs](https://docs.budibase.com/docs/accessing-minio) that minIO could be the issue and followed the instructions to enable it. 
When it didn't work, I followed [another thread](https://github.com/Budibase/budibase/discussions/5084) to add the ports in the docker-compose.yaml file. Lastly I changed the .env file port but still the attachments are not working. I added firewall rules for ports 9000 and 9001 using UFW to try and access the minIO interface with no luck as well. I think I've exausted my options for solutions that have been discussed by others.

Edit: I added a text field and an attachment field to a table in the internal database. Instead of trying to add a record with both fields at once I entered just the text part and managed to save it. Afterwards I did an edit and added the atachment and the record vanished after saving.

Can anyone help?

**Comment by melohagan** [comment]
Hey @nunopoco 

I wasn't able to reproduce this. Could you provide some screenshots to highlight what is going wrong?

If you look at the console and network tabs of the browser (Fn+F12 on Chrome) do you see any error messages when you try to save an attachment?

**Comment by nunopoco** [comment]
Hi @melohagan thanks for helping me with this.
I've created a video showing that the files are being uploaded, it's just the UI that is not showing the records.
Check it out: https://nuno.dubb.com/v/ZIuB4O
Sorry for the background noise.

**Reply by melohagan** [reply]
Hey @nunopoco Thanks for providing the video. I noticed when you clicked on the edit button that a JS error appeared in your console. Probably not related but worth pointing out.

Would you be able to share the app export to support@budibase.com ?

I did follow your steps but not seeing an issue myself. Maybe it's an nginx issue? Is your app URL in the prod nginx conf:
<img width="865" alt="Screenshot 2022-09-23 at 14 02 13" src="https://user-images.githubusercontent.com/101575380/191966166-01ba0438-b429-4ea1-be57-ceac3a074bca.png">

**Reply by nunopoco** [reply]
I removed the extension creating the JS error and this one persists.

Also, it seems nginx is not installed.
I tried running "systemctl status nginx" and received "Unit nginx.service could not be found." as an answer.
Should I install it? Is there any step-by-step instructions on how to do it and then configure Budibase to use it?
I sent the app export to the support email, as requested.

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @nunopoco Thanks for the app export, I was able to recreate your issue.

It's not a proxy issue. Unfortunately it looks like your app is corrupt:
![Screenshot 2022-09-26 at 10 13 45](https://user-images.githubusercontent.com/101575380/192239177-7e16e59e-84a2-4a32-87d9-cb37d396200a.png)

I found that there was a conflict in your CouchDB.

1. Go to `<url>:4005/_utils`
2. Login to the couchdb instance with username/password: `budibase` : `budibase`

3. Find the database for your app. The app ID can be found in the URL of your Budibase data screen.
![Screenshot 2022-09-26 at 10 32 59](https://user-images.githubusercontent.com/101575380/192243311-d8bcae9e-f7d9-474b-81a1-20ea984372bd.png)

4. Click on the `_design/database` table
![Screenshot 2022-09-26 at 10 33 57](https://user-images.githubusercontent.com/101575380/192243510-c474626a-8285-4310-abd7-492b5911a7b2.png)

5. Click on `Conflicts`
![Screenshot 2022-09-26 at 10 34 44](https://user-images.githubusercontent.com/101575380/192243675-a65a2a60-34c8-4101-b647-6e5a94e7e455.png)

6. Click `Select as Winner` and click on `Delete revisions`. A message should pop up stating conflicts resolved.
![Screenshot 2022-09-26 at 10 35 38](https://user-images.githubusercontent.com/101575380/192243883-7e172449-2520-4e63-8a4d-1a82b713744d.png)

![Screenshot 2022-09-26 at 10 36 28](https://user-images.githubusercontent.com/101575380/192244051-e32c5b0b-b55f-459d-a19e-67bfe478e8ed.png)

7. You should now see your attachments!
![Screenshot 2022-09-26 at 10 38 03](https://user-images.githubusercontent.com/101575380/192244383-fcc37ada-ea8f-4bff-ac1a-5bbe55beae93.png)

**Reply by nunopoco** [reply]
> * Go to `<url>:4005/_utils`

I tried accesing it but it doesn't load.
I used ufw to allow port 4005 but nothing changed.
Any tips on how I might be able to reach it?

**Reply by melohagan** [reply]
Are your running via docker compose? You should be able to set the port forwarding like so:

```yaml
couchdb-service:
    ports:
      - "4005:5984"
```

In addition, if you are running Budibase on an ARM machine (e.g. M1 Macbook) then you'll want to add this line to the couchdb-service too:
`platform: linux/amd64`

**Reply by nunopoco** [reply]
> Are your running via docker compose? You should be able to set the port forwarding like so:
>
> ```yaml
> couchdb-service:
>     ports:
>       - "4005:5984"
> ```

I altered the docker-compose.yaml file and added these intructions but I don't know how to restart couchdb service, so I restarted the server instead. Still not able to access it.

> In addition, if you are running Budibase on an ARM machine (e.g. M1 Macbook) then you'll want to add this line to the couchdb-service >too: `platform: linux/amd64`

I just deployed to DigitalOcean using the wizard in BB cloud, didn't do anything else, just started working right away.
Should I export the app, delete the droplet, run the install wizard again and re-import the app?

**Reply by jonnymccullagh** [reply]
@nunopoco after making a change to the `docker-compose.yaml` file you can stop the containers with:
```
docker-compose stop
```
Then start them again (but requesting that the changed config is re-read):
```
docker-compose up -d --build
```
Check the output from docker-compose:
```
docker-compose ps
```
check the Ports column for the couchdb container. If you have requested to forward port 4005 from the server into port 5984 of the container you should see this:
```
0.0.0.0:4005->5984/tcp
```
At this point you should be able to open a web browser using the IP address of the droplet and the port you forwarded to couchdb (4005) e.g.:
http://111.222.111.222:4005/_utils/#login

If you have problems with this please post the output of the `docker-compose ps` command. 
Also remember to remove the couchdb port forwarding after fixing the problem. You won't want to keep your database exposed.

**Reply by jonnymccullagh** [reply]
Video now added on Budibase docs page demonstrating this:
https://docs.budibase.com/docs/digitalocean
